tech/post/2017-11-06-Kubernetes-Taints&Tolerations.md

```diff
@@ -0,0 +1,302 @@
+# Taints and Tolerations
+
+## 1. Introducation
+
+Taints and tolerations work together to ensure that pods are not scheduled onto inappropriate nodes.
+One or more taints are applied to a node; this marks that the node should not accept any pods that do not tolerate the taints.
+Tolerations are applied to pods, and allow the pods to schedule onto nodes with matching taints.
+
+![](../images/2017-11-06-Kubernetes-Taints&Tolerations/Concept.png)
+
+Node affinity is a property of pods that attracts them to a set of nodes.
+Taints are the opposite and they allow a node to repel a set of pods.
+If Node affinity could be called by ```White List```, Taints could be called by ```Black List```.
+
+## 2. Operations
+
+### Basis
+
+Taints are defined for Kubernetes Cluster nodes in the ```annotations```,
+and Tolerations are specified for Kubernetes pods in the ```PodSpec```.
+
+The Scheduler will schedule the pods by lots of rules, including below:
+
+* PodToleratesNodeTaints: one of the predicates rules.
+
+  Judge whether Pod satisfy the conditions of node tolerance.

```

Comment:
Judge whether the Pod satisfies or Judge whether Pods satisfy?

Reply:
Done